powersimdata/network/model.py

```diff
@@ -34,75 +37,59 @@ def __init__(self, model, interconnect=None):
+        raise TypeError("area must be a str")
 
     if area_type is not None and not isinstance(area_type, str):
-        raise TypeError("'area_type' should be either None or str.")
+        raise TypeError("area_type must be either None or str")
+
+    division = [a for a in mappings if "abbr" in a][0].split("_")[0]
+    area2loadzone = OrderedDict(
+        {
+            f"{division}": lambda x: zones[f"{division}2loadzone"][x],
+            "loadzone": lambda x: zones["loadzone"].intersection({x}),
+            f"{division}_abbr": lambda x: zones["abv2loadzone"][x],
+            "interconnect": lambda x: zones["interconnect2loadzone"][x],
+            "all": lambda _: zones["loadzone"],
+        }
+    )
+
     if area_type:
-        if area_type == "loadzone" and "loadzone" in mappings:
-            if area in zones["loadzone"]:
-                loadzone_set = {area}
-            else:
-                raise_invalid_area(area_type)
-        elif area_type == "state" and "state" in mappings:
-            if area in zones["abv2state"].values():
-                loadzone_set = zones["state2loadzone"][area]
-            else:
-                raise_invalid_area(area_type)
-        elif area_type == "state_abbr" and "state_abbr" in mappings:
-            if area in zones.abv2state:
-                loadzone_set = zones["state2loadzone"][zones["abv2state"][area]]
-            else:
-                raise_invalid_area(area_type)
-        elif area_type == "interconnect" and "interconnect" in mappings:
-            if area in zones["interconnect2loadzone"]:
-                loadzone_set = zones["interconnect2loadzone"][area]
-            else:
-                raise_invalid_area(area_type)
-        else:
-            print(f"{area_type} is incorrect. Available area_types are: {mappings}.")
-            raise ValueError("Invalid area_type")
+        if area_type not in mappings:
+            raise ValueError(f"Invalid area type. Choose among {' | '.join(mappings)}")
+        if area not in zones[area_type]:
+            raise ValueError("Invalid area / area_type combination")
+        loadzone = area2loadzone[area_type](area)
     else:
-        if "state" in mappings and area in zones["abv2state"].values():
-            loadzone_set = zones["state2loadzone"][area]
-        elif "loadzone" in mappings and area in zones["loadzone"]:
-            loadzone_set = {area}
-        elif "state" in mappings and area in zones["abv2state"]:
-            loadzone_set = zones["state2loadzone"][zones["abv2state"][area]]
-        elif "interconnect" in mappings and area in zones["interconnect2loadzone"]:
-            loadzone_set = zones["interconnect2loadzone"][area]
-        elif area == "all":
-            loadzone_set = zones["loadzone"]
-        else:
-            print("%s is incorrect." % area)
+        zones["all"] = "all"
+        loadzone = set().union(
+            *(area2loadzone[a](area) for a in area2loadzone if area in zones[a])
+        )
```

Comment:
Brilliant! I didn't realize `state` is always the superset of load zones among the mappings of a unique keyword, i.e. if `area='North Carolina'`, we first evaluate it as a state, which gives two load zones `North Carolina and Western North Carolina`, then we evaluate it as a loadzone, which gives `North Carolina` only, finally the union won't change the result, same applies for `Texas` since $ERCOT\ \subseteq \ Texas$ in terms of load zones. Hence the if/else logic in the original implementation is equivalent to taking the union among the results of all searches. The only thing is in this way we wouldn't need `area2loadzone` to be an `OrderedDict` anymore, right?

Reply:
> $ERCOT\ \subseteq \ Texas$

Wow, fancy!

Reply:
I think we need an `OrderedDict` here to guarantee we loop in the right order as described in the docstring. Not sure I completely understand your question.

Reply:
We are going to union the results no matter in what order it loops through, right? Previously I thought this might end up with different behavior from the original implementation, but thinking it twice I found the behavior is the same since we don't have a loadzone with the same name as a state and defines a larger area.

Reply:
I understand now. I did not think too much about it, I followed the same search sequence than in the previous implementation. What you say makes sense and I will use a standard dict.

Reply:
Done